garak/generators/mistral.py

```diff
@@ -47,11 +87,170 @@ def _call_model(
         self, prompt: Conversation, generations_this_call=1
     ) -> List[Message | None]:
         messages = self._conversation_to_list(prompt)
-        chat_response = self.client.chat.complete(
-            model=self.name,
-            messages=messages,
-        )
-        return [Message(chat_response.choices[0].message.content)]
+        
+        try:
+            logging.debug(f"Making Mistral API call with model: {self.name}")
+            # Build API parameters, excluding None values
+            api_params = {
+                "model": self.name,
+                "messages": messages,
+            }
+            
+            # Add all non-None parameters using dictionary comprehension
+            api_params.update({
+                api_param: getattr(self, attr_name)
+                for attr_name, api_param in self._PARAM_MAPPING.items()
+                if getattr(self, attr_name, None) is not None
+            })
+            
+            # Handle special case: top_k fallback to top_p
+            if self.top_k is not None and self.top_p is None:
+                api_params["top_p"] = self.top_k
+                
+            chat_response = self.client.chat.complete(**api_params)
+            logging.debug(f"Mistral API call successful, processing response")
+            
+            # Handle both string content and list of chunks (with thinking sequences)
+            content = chat_response.choices[0].message.content
+            if isinstance(content, list):
+                logging.debug(f"Mistral returned chunked content with {len(content)} chunks")
+                # Extract text chunks and filter out thinking chunks
+                text_parts = []
+                for chunk in content:
+                    if isinstance(chunk, dict):
+                        chunk_type = chunk.get('type', 'unknown')
+                        if chunk_type == 'text':
+                            text_parts.append(chunk.get('text', ''))
+                        elif chunk_type == 'thinking':
+                            logging.debug(f"Filtered out thinking chunk: {chunk.get('thinking', '')[:100]}...")
+                        else:
+                            logging.warning(f"Unknown chunk type: {chunk_type}")
+                content = ''.join(text_parts)
+                logging.debug(f"Final extracted content length: {len(content)}")
+            
+            return [Message(content)]
+            
+        except Exception as e:
+            # If we get a Pydantic validation error, try to make a raw HTTP request
```

Comment:
This is not really something a generator utilizing a `client` package should do.  If the user is expecting to test a target in violation of the client provided spec then the `RestGenerator` should be used.